packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -86,7 +86,13 @@ ComponentNodeManager.create = function ComponentNodeManager_create(renderNode, e
 
     if (!layout) { return; }
 
-    let fragmentReason = layout.meta.fragmentReason;
+    let fragmentReason;
+    if (layout.meta === undefined) {
+      assert('Meta key missing from layout. This usually happens when there is no template for a Component, and the Component has overrided the layout key. Please rename the property to something other than "layout".');
+    } else {
+      fragmentReason = layout.meta.fragmentReason;
+    }
```

Comment:
Since asserts are stripped from prod builds, they should contain the test:

```js
    assert('Meta key missing from layout. This usually happens when there is no template for a Component, and the Component has overrided the layout key. Please rename the property to something other than "layout".', layout.meta !== undefined);
    let fragmentReason = layout.meta.fragmentReason;
```